std/os/linux.zig

```diff
@@ -1607,6 +1607,32 @@ pub const dirent64 = extern struct {
     d_name: u8, // field address is the address of first byte of name https://github.com/ziglang/zig/issues/173
 };
 
+pub const SS_ONSTACK = 1;
+pub const SS_DISABLE = 2;
+pub const SS_AUTODISARM = 1 << 31;
+
+pub const SIGSTKSZ = switch (builtin.arch) {
+    .i386, .x86_64, .arm => 8192,
+    .aarch64 => 16384,
+    else => @compileError("SIGSTKSZ not defined for this architecture"),
+};
+
+pub const MINSIGSTKSZ = switch (builtin.arch) {
+    .i386, .x86_64, .arm => 2048,
+    .aarch64 => 6144,
+    else => @compileError("MINSIGSTKSZ not defined for this architecture"),
```

Comment:
Doesn't this make things fail on all non-listed architectures?

Reply:
Also we can grab all the values from the kernel, this search is useful: https://elixir.bootlin.com/linux/latest/ident/MINSIGSTKSZ

Reply:
Note that so far all the per-arch things are in their own files (see e.g. linux/arm64.zig)

Reply:
> Doesn't this make things fail on all non-listed architectures?

Yes? IMO it's better to be explicit about missing definitions.

> Also we can grab all the values from the kernel, this search is useful

That's not so easy, sometimes different libcs muddle the waters by changing that value. E.g. The aarch64 value is greater than the kernel-reported one. This begs another question, do we want to stay true to the kernel or try to paper over the kernel+libc differences?

> Note that so far all the per-arch things are in their own files (see e.g. linux/arm64.zig)

See point 1, `pub use` masks the missing definitions unless you actually try to use it.

Reply:
> E.g. The aarch64 value is greater than the kernel-reported one

Is it? I thought it was only ia64 that was weird https://elixir.bootlin.com/linux/latest/source/arch/ia64/include/uapi/asm/signal.h#L95

> do we want to stay true to the kernel or try to paper over the kernel+libc differences?

Usually I'd say "stay true to the kernel", but for the specific case of a "minimum space required", I'd take the larger of the two options (which in the case of ia64, is userspace)

Reply:
> Is it? I thought it was only ia64 that was weird

See for yourself, here are [glibc](https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/unix/sysv/linux/aarch64/bits/sigstack.h;h=14e4ed86337829dec987e38748b962304b60e5b9;hb=HEAD) and [musl](https://git.musl-libc.org/cgit/musl/tree/arch/aarch64/bits/signal.h) definitions.

Reply:
> See for yourself, here are [glibc](https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/unix/sysv/linux/aarch64/bits/sigstack.h;h=14e4ed86337829dec987e38748b962304b60e5b9;hb=HEAD) and [musl](https://git.musl-libc.org/cgit/musl/tree/arch/aarch64/bits/signal.h) definitions.

Interesting. Is this because libc reserves some stack space in signal handling for itself?
If that's the case, inline with my [comment here](https://github.com/ziglang/zig/pull/2527#issuecomment-495469872), `std/os/linux.zig` should have the kernel definition, `std/c/linux.zig` should have the libc definition.

Reply:
> Is this because libc reserves some stack space in signal handling for itself?

[Here's](https://git.musl-libc.org/cgit/musl/commit/arch/aarch64/bits/signal.h?id=d5a5045382315e36588ca225889baa36ed0ed38f) the rationale:
>  since the default (SIGSTKSZ) has always been 6k larger than the minimum, it is also increased to maintain the 6k usable by the signal handler. this happens to be able to store one pathname buffer and should be sufficient for calling any function in libc that doesn't involve conversion between floating point and decimal representations. 

Any number is fine as long as you're able to satisfy the requirements imposed by the kernel.

I'm also interested in hearing your ideas on how to define this in `c.zig`, should `stack_t` be redefined there? Should it reference the definition in `linux.zig`?

Reply:
>  the rationale:

So sounds like libc just adds some extra space "just in case": I would go with the kernel definition.
Though I would be okay adding a separate definition to `std/c/linux.zig` so that calling C libraries that assume enough space for a pathname buffer don't fail.



> I'm also interested in hearing your ideas on how to define this in `c.zig`, should `stack_t` be redefined there? Should it reference the definition in `linux.zig`?

Yes and yes. `stack_t` would fall under the condition I outlined of:
> If libc defines a function that e.g. "takes a pointer to a native `struct sockaddr`" then you may use type definitions from `std/os/*`

Reply:
> So sounds like libc just adds some extra space "just in case": I would go with the kernel definition.

The kernel prescribes the minimum amount of stack you need for an _empty_ handler to execute, if you want to do anything remotely useful you need more space than that.

I agree on picking the kernel definition here, let's report the bare minimum and then have the user pick a good value.

Reply:
Yep. The user should really be using the kernel minimum plus `@stack_size(their_handler)` (see  https://github.com/ziglang/zig/issues/157 )